playground/infrastructure/helper.py

```diff
@@ -44,29 +53,187 @@ def find_examples(work_dir: str) -> List[Example]:
     """
     # TODO [BEAM-13267] Implement
     pass
+
+
+def get_tag(filepath):
+    """
+    Parse file by filepath and find beam tag
+
+    Args:
+        filepath: path of the file
+
+    Returns:
+        If file contains tag, returns tag as a map.
+        If file doesn't contain tag, returns None
+    """
+    add_to_yaml = False
+    yaml_string = ""
+
+    with open(filepath) as parsed_file:
+        lines = parsed_file.readlines()
+
+    for line in lines:
+        line = line.replace("// ", "").replace("# ", "")
+        if add_to_yaml is False:
+            if line == BEAM_PLAYGROUND_TITLE:
+                add_to_yaml = True
+                yaml_string += line
+        else:
+            yaml_with_new_string = yaml_string + line
+            try:
+                yaml.load(yaml_with_new_string, Loader=yaml.SafeLoader)
+                yaml_string += line
+            except Exception:
+                break
+
+    if add_to_yaml:
+        tag_object = yaml.load(yaml_string, Loader=yaml.SafeLoader)
+        return tag_object[BEAM_PLAYGROUND]
+
+    return None
+
+
+def _get_example(filepath: str, filename: str) -> Example:
+    """
+    Return an Example by filepath and filename.
+
+    Args:
+         filepath: path of the example's file.
+         filename: name of the example's file.
+
+    Returns:
+        Parsed Example object.
+    """
+    name = _get_name(filename)
+    sdk = _get_sdk(filename)
+    with open(filepath) as parsed_file:
+        content = parsed_file.read()
+
+    return Example(name, "", sdk, filepath, content, "", STATUS_UNSPECIFIED)
+
+
+def _validate(tag: dict, categories_path: str) -> bool:
+    """
+    Validate all tag's fields
+
+    Validate that tag contains all required fields and all fields have required format.
+
+    Args:
+        tag: beam tag to validate
+        categories_path: path to the file with all supported categories.
+
+    Returns:
+        In case tag is valid, True
+        In case tag is not valid, False
+    """
+    valid = True
+    if tag.get("name") is None:
+        logging.error("tag doesn't contain name field: " + tag.__str__())
+        valid = False
+    if tag.get("description") is None:
+        logging.error("tag doesn't contain description field: " + tag.__str__())
+        valid = False
+    if tag.get("multifile") is None:
+        logging.error("tag doesn't contain multifile field: " + tag.__str__())
+        valid = False
+    multifile = tag.get("multifile")
+    if str(multifile).lower() not in ["true", "false"]:
+        logging.error("tag's field multifile is incorrect: " + tag.__str__())
+        valid = False
+    if tag.get("categories") is None:
+        logging.error("tag doesn't contain categories field: " + tag.__str__())
+        valid = False
+    categories = tag.get("categories")
+    if type(categories) is not list:
+        logging.error("tag's field categories is incorrect: " + tag.__str__())
+        valid = False
```

Comment:
If `categories` is actually a list, we can also check if the listed categories are in supported ones.

Reply:
each `category` are being checked